things/api/grpc/endpoint_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package grpc_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/absmach/magistrala"
+	"github.com/absmach/magistrala/auth"
+	"github.com/absmach/magistrala/internal/apiutil"
+	"github.com/absmach/magistrala/pkg/errors"
+	grpcapi "github.com/absmach/magistrala/things/api/grpc"
+	"github.com/absmach/magistrala/things/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const port = 7000
+
+var (
+	thingID   = "testID"
+	channelID = "testID"
+	invalid   = "invalid"
+	valid     = "valid"
+)
+
+func startGRPCServer(svc *mocks.Service, port int) {
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+	if err != nil {
+		panic(fmt.Sprintf("failed to obtain port: %s", err))
+	}
+	server := grpc.NewServer()
+	magistrala.RegisterAuthzServiceServer(server, grpcapi.NewServer(svc))
+	go func() {
+		if err := server.Serve(listener); err != nil {
+			panic(fmt.Sprintf("failed to serve: %s", err))
+		}
+	}()
+}
+
+func TestAuthorize(t *testing.T) {
+	svc := new(mocks.Service)
+	startGRPCServer(svc, port)
+	authAddr := fmt.Sprintf("localhost:%d", port)
+	conn, _ := grpc.Dial(authAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	client := grpcapi.NewClient(conn, time.Second)
+
+	cases := []struct {
+		desc         string
+		authorizeReq *magistrala.AuthorizeReq
+		res          *magistrala.AuthorizeRes
+		thingID      string
+		authorizeErr error
+		err          error
+		code         codes.Code
+	}{
+		{
+			desc: "authorize successfully",
+			authorizeReq: &magistrala.AuthorizeReq{
+				Subject:     thingID,
+				SubjectKind: auth.ThingsKind,
+				Permission:  valid,
+				SubjectType: auth.ThingType,
+				Object:      channelID,
+				ObjectType:  auth.GroupType,
+			},
+			thingID: thingID,
+			res:     &magistrala.AuthorizeRes{Authorized: true, Id: thingID},
+			err:     nil,
+		},
+		{
+			desc: "authorize with invalid id",
+			authorizeReq: &magistrala.AuthorizeReq{
+				Subject:     invalid,
+				SubjectKind: auth.ThingsKind,
+				Permission:  "publish",
+				SubjectType: auth.ThingType,
+				Object:      channelID,
+				ObjectType:  auth.GroupType,
+			},
+			res: &magistrala.AuthorizeRes{Authorized: false, Id: ""},
```

Comment:
The string will default to empty and the boolean will default to false. This applies to other cases

```suggestion
			res: &magistrala.AuthorizeRes{},
```